CSharpMath.Forms.Tests/ButtonTests.cs

```diff
@@ -23,28 +21,27 @@ public class ButtonTests {
       Assert.Equal(mathButton.TextColor, newColor);
       Assert.Equal(latexContent, textButton.Content.LaTeX);
     }
-    [Fact]
-    public void AllMathInputButtonsHaveLatexContent() {
-      foreach (var mathInputButton in TheMathInputButtons) {
-        Assert.True(mathInputButton.Content != null && !string.IsNullOrEmpty(mathInputButton.Content.LaTeX));
-      }
+    [Theory]
+    [ClassData(typeof(TheMathInputButtons))]
+    public void AllMathInputButtonsHaveLatexContent(MathInputButton mathInputButton) {
+      Assert.False(string.IsNullOrEmpty(mathInputButton.Content?.LaTeX));
     }
-    [Fact]
-    public void MathInputButtonsHaveBlackTextColorByDefault() {
-      foreach (var mathInputButton in TheMathInputButtons) {
-        // At the time of writing this test, vphatom = @"{\color{#00FFFFFF}{|}}" (Xamarin.Forms.Color.Transparent = "#00FFFFFF") is used as there is no \vphantom command yet.
-        // As soon as \vphantom has been implemented, the call .Replace(LatexHelper.vphantom, "") can be removed.
-        Assert.DoesNotContain(@"\color", mathInputButton.Content!.LaTeX!.Replace(LatexHelper.vphantom, ""));
-        Assert.Equal(Color.Black, mathInputButton.TextColor);
-      }
+    [Theory]
+    [ClassData(typeof(TheMathInputButtons))]
+    public void MathInputButtonsHaveBlackTextColorByDefault(MathInputButton mathInputButton) {
+      // At the time of writing this test, vphatom = @"{\color{#00FFFFFF}{|}}" (Xamarin.Forms.Color.Transparent = "#00FFFFFF") is used as there is no \vphantom command yet.
+      // As soon as \vphantom has been implemented, the call .Replace(LatexHelper.vphantom, "") can be removed.
+      Assert.DoesNotContain(@"\color", mathInputButton.Content.NotNull().LaTeX.NotNull().Replace(LatexHelper.vphantom, ""));
+      Assert.Equal(Color.Black, mathInputButton.TextColor);
     }
-    [Fact]
-    public void MathInputButtonsHaveTransparentBackgroundByDefault() {
-      foreach (var mathInputButton in TheMathInputButtons) {
-        Assert.Equal(Color.Transparent, mathInputButton.BackgroundColor);
-      }
+    [Theory]
+    [ClassData(typeof(TheMathInputButtons))]
+    public void MathInputButtonsHaveTransparentBackgroundByDefault(MathInputButton mathInputButton) {
+      Assert.Equal(Color.Transparent, mathInputButton.BackgroundColor);
+    }
+    public class TheMathInputButtons : TestHelpers.ComplexClassData<MathInputButton> {
```

Comment:
I suggest `MemberData` instead of `ClassData` for less code.

Reply:
Also, should use `MathKeyboardInput` as parameters since only xunit-serializable types will result in individual test cases instead of one single test case with multiple results.